### DISCUSSION TITLE
Get the topmost actor that is notified by pointerup event

### DISCUSSION THREAD

**Discussion by LokiMidgard** [excaliburjs/Excalibur#1899] [Q&A] [answered]
If you have Multiple actor's that share the same position and subscribe to the same event. It is fired on every actor.

You can stop it by setting the bubbling to false, but since the bottom most actor is notified at first, this is the opposite of what I need.

Is there any way besides manually checking if there is any actor above this current one to prevent this?

**Comment by eonarheim** [comment, marked as answer]
@LokiMidgard we spoke offline to the team and we are thinking this is a bug/undesired behavior with event bubbling.

We think it may make more sense for the events to bubble from top-most z-index to bottom-most z-index? That way you'd be able to stop the bubbling in a more expected way?

Let me know and I'll make a bug :+1:

**Reply by LokiMidgard** [reply]
I agree, that in the most cases the top most is the desired one. But it may be necessary to support both directions.

**My proposal would be:**

To mark the event if more actors will be processed. You could use the bubbling itself. Currently bubbling on the event is always true and you can set it to false if you want to prevent further processing. The event could use the same filed to notify if additional actors will be processed. If the filed is `true` there are more Actors if it is `false` this is the last actor.

---------------------------------------------------
*side note*  
Coming from C# XAML I associate bubbling with from bottom to top and from top to bottom with tunneling.

**Reply by eonarheim** [reply]
I've written up a bug 👍 https://github.com/excaliburjs/Excalibur/issues/1922
